include/zephyr/dt-bindings/memory-attr/memory-attr.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_MEM_ATTR_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_MEM_ATTR_H_
+
+/* Ideally we'd generate this enum to match what's coming out of the YAML,
+ * however, we dont have a good way to know how to name such an enum from the
+ * generation point of view, so for now we just hand code the enum.  This
+ * enum is expected to match the order in the yaml (dts/bindings/base/zephyr,memory-attr.yaml)
+ */
+
+enum dt_memory_attr {
+	DT_MEMORY_ATTR_RAM,
+	DT_MEMORY_ATTR_RAM_NOCACHE,
+	DT_MEMORY_ATTR_FLASH,
+	DT_MEMORY_ATTR_PPB,
```

Comment:
ppb?  maybe add a comment, since I've got no clue what PPB is ;)

Reply:
@galak replying here also for the other comment.

Agree, naming and description is a big issue here. @manuargue will be taking care of that soon, see https://github.com/zephyrproject-rtos/zephyr/pull/60049#pullrequestreview-1528662682 and https://github.com/zephyrproject-rtos/zephyr/pull/60049#issuecomment-1635832451 but I think we are forced to live with it for a little while, but cleaning up is already planned I swear :)